packages/php-wasm/compile/Dockerfile

```diff
@@ -1036,16 +1050,16 @@ RUN \
         # environment, based on the `RuntimeName` variable, which is an argument to the
         # wrapper function.
         /root/replace.sh $'s/ENVIRONMENT_IS_([A-Z]+)\s*=\s*(true|false)/ENVIRONMENT_IS_$1=RuntimeName==="$1"/g' /root/output/php.js && \
-        /root/replace.sh 's/var ENV\s*=\s*\{\}/var ENV = PHPLoader.ENV || {}/g' /root/output/php.js && \
+        /root/replace.sh 's/var ENV\s*=\s*\{\}/var ENV = PHPLoader.ENV || {}/g' /root/output/php.js; \
```

Comment:
Why replace `&&` with `;` here? For context, `replace.sh` is designed to return an error code when the pattern string can't be found.

Reply:
@adamziel 

That may be an artifact of some testing I was doing. However, going forward, what would you think of using `SHELL ["/bin/bash", "-euc", -"o pipefail"]` at the start of the file? This would cause the process to immediately stop and report on ANY process error or unset variable.

If we REALLY needed to accept a non-zero exit code, we could use the notation `command || true` to ensure the shell always receives an exit code of 0.

Reply:
@seanmorris Lovely idea! It should work nicely as a follow-up PR as I worry we may need to add quite a few of `|| true` which could spin into a longer discussion about the benefits, coding styles, alternatives etc. Let's just stick `&&` for now unless there's a specific reason to switch to `;`.

Reply:
@adamziel We could start to implement this by adding `set -eu -o pipefail` as the first line to individual `RUN` commands. In this case the effect will only apply to that particular `RUN`, and once we've got good saturation on that front, migrating to `SHELL ["/bin/bash", "-euc", -"o pipefail"]` will be trivial.

Reply:
@adamziel I noticed that the `if` statements are suffixed with `;`'s instead of `&&`, i.e. `fi; \` rather than `fi && \`, is this expected? I think some failures could be missed in this case, so I added the `set` command from above to the start of the section.

https://github.com/WordPress/wordpress-playground/blob/93bf38f47be4ec5ed08b003e145c4df37eb799c8/packages/php-wasm/compile/Dockerfile#L1061

Reply:
Sounds good! Let's treat these two RUN commands like this in this PR, and work out the general Dockerfile style later on.

> I noticed that the if statements are suffixed with ;'s instead of &&, i.e. fi; \ rather than fi && \, is this expected?

That's a bug. I vaguely remember Docker or bash insisted that using `&&` was a syntax error and I just went for `;`.